lib/configuration/variables/index.js

```diff
@@ -15,6 +15,7 @@ const defaultSources = {
   opt: require('./sources/opt'),
   self: require('./sources/self'),
   strToBool: require('./sources/str-to-bool'),
+  sls: require('./sources/instance-dependent/get-sls')(),
```

Comment:
Ah, I see, we can resolve `stage` but e.g. if `instanceId` would be used then it won't work once again - at some point we will need to address the issue with variables resolution in interactive flow in a better way because the recent fix from me only addressed the `sls` source and not other sources that might be added/used in templates

Reply:
Yes, main problem with other sources is that due to how AWS credential resolution works, they depend on `serverless` instance.

Once we fix https://github.com/serverless/serverless/issues/9290, we'd be able to do it cleanly here.